underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -270,50 +369,61 @@ public InputStream open(String path) throws IOException {
   }
 
   /**
-   * Each path is checked both for leading "/" and ending "/". Leading "/" is removed, and "/" is
-   * added at the end if not present.
+   * A trailing {@link SwiftUnderFileSystem#PATH_SEPARATOR} is added if not present.
    *
    * @param path URI to the object
-   * @return qualified path
+   * @return folder path
    */
-  private String makeQualifiedPath(String path) {
-    if (!path.endsWith("/")) {
-      path = path + "/";
-    }
-    if (path.startsWith("/")) {
-      path = path.substring(1);
-    }
-    return path;
+  private String addFolderSuffixIfNotPresent(String path) {
+    return PathUtils.normalizePath(path, PATH_SEPARATOR);
   }
 
   /**
    * @inheritDoc
    *
-   * @param src the source file or folder name
-   * @param dst the destination file or folder name
+   * @param source the source file or folder name
+   * @param destination the destination file or folder name
    * @return true if succeed, false otherwise
    * @throws IOException if a non-Alluxio error occurs
    */
   @Override
-  public boolean rename(String src, String dst) throws IOException {
-    String strippedSrcPath = stripPrefixIfPresent(src);
-    String strippedDstPath = stripPrefixIfPresent(dst);
-    if (exists(src) && copy(strippedSrcPath, strippedDstPath)) {
-      return delete(src, true);
+  public boolean rename(String source, String destination) throws IOException {
```

Comment:
Every `rename` does at least a call to `exists` for the source and destination and `isDirectory()`, which all could be expensive.

On top of that, `rename` is called recursively, which will eventually call all those RPCs again.

Are all those "checks" necessary? What would happen with `copy()` or `delete() if the source or destination do not exist? Also, concurrent clients could alter source and destination anyways, so even if the `exists` checks are done, I don't think it will guarantee that they `exist` when it comes time to `copy` or `delete`.

What do you think?